src/QueryBuilderParser/JoinSupportingQueryBuilderParser.php

```diff
@@ -95,28 +95,51 @@ private function buildSubclauseQuery($query, $rule, $value, $condition)
 
         $not = array_key_exists('not_exists', $subclause) && $subclause['not_exists'];
 
-        // Create a where exists clause to join to the other table, and find results matching the criteria
-        $query = $query->whereExists(
-            /**
-             * @param Builder $query
-             */
-            function(Builder $query) use ($subclause) {
-
-                $q = $query->selectRaw(1)
-                    ->from($subclause['to_table'])
-                    ->whereRaw($subclause['to_table'].'.'.$subclause['to_col']
-                        .' = '
-                        .$subclause['from_table'].'.'.$subclause['from_col']);
-
-                if (array_key_exists('to_clause', $subclause)) {
-                    $q->where($subclause['to_clause']);
-                }
-
-                $this->buildSubclauseInnerQuery($subclause, $q);
-            },
-            $condition,
-            $not
-        );
+	    if ( $not ) {
```

Comment:
One thing that I am cautious of is that this code _looks like_ it will change the behavior for other users of this class.

Can we separate out the code inside the `if() {...}` block into two separate functions to ensure that the length of this function (`buildSubclauseQuery`) is not too long?